workflow/controller/controller.go

```diff
@@ -161,7 +158,7 @@ func NewWorkflowController(ctx context.Context, restConfig *rest.Config, kubecli
 		configController:           config.NewController(namespace, configMap, kubeclientset, config.EmptyConfigFunc),
 		workflowKeyLock:            syncpkg.NewKeyLock(),
 		cacheFactory:               controllercache.NewCacheFactory(kubeclientset, namespace),
-		eventRecorderManager:       events.NewEventRecorderManager(kubeclientset),
+		EventRecorderManager:       events.NewEventRecorderManager(kubeclientset),
```

Comment:
Rather than expose this, please can you just create a brand new one for the leader election? This violates inversion of control principle.